### PR TITLE
accept webroot shell parameter

### PR DESCRIPTION
**Pull request by bancer**
resolves https://github.com/cakephp/cakephp/issues/9374

### CODE REVIEW

**Comment by markstory** [comment]
The existing [tests are failing](https://travis-ci.org/cakephp/cakephp/jobs/174070632) that will need to be fixed before this can be merged.

File: lib/Cake/Test/Case/Console/ShellDispatcherTest.php [hunk 1]
```diff
@@ -191,17 +214,24 @@ public function testParseParams() {
...
 		$this->assertEquals($expected, $Dispatcher->params);
+	}
+
+/**
+ * testParseParams method
+ *
+ * @return void
+ */
+	public function testParseParams() {
+		$Dispatcher = new TestShellDispatcher();
```
Comment (markstory) [inline lib/Cake/Test/Case/Console/ShellDispatcherTest.php]: Thank you for splitting up the mega test method. :clap:

Comment (bancer) [inline lib/Cake/Test/Case/Console/ShellDispatcherTest.php]: I did it only partially though.

Comment (markstory) [inline lib/Cake/Test/Case/Console/ShellDispatcherTest.php]: It is better than before which is what counts :+1:

**Pull request merged**